ProjectTraning/Task_20_07/FridgeRefreg.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ProjectTraning.Task_20_07
+{
+    public class FridgeRefreg
+    {
+        public event FridgeDelegate Fridge;
+
+        public Enum Status { get; set; } = ModesofOperation.On;
+
+        public Enum MainDor { get; set; } = ModesofOperation.Close;
+
+        public Enum SecondDor { get; set; } = ModesofOperation.Close;
+
+        protected virtual void OnFridge(FridgeEventArgs e)
+        {
+            this.Fridge?.Invoke(this, e);
+        }
+
+        public string StatusOfFridge()
+        {
+            return $"Status - {Status} \nMainDor - {MainDor} \nSecond Dor - {SecondDor}"; 
+        }
+
+        public void FridgeOn()
+        {          
+            Status = ModesofOperation.On;
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.Write("\nХолодильник принимает статус: ");
```

Comment:
The state should be passed through the custom event args.
And the manager should tell us if event was invoked, not fridge itself.

Reply:
-1 to customEventArgs